setting_test.go

```diff
@@ -6,6 +6,20 @@ import (
 	"time"
 )
 
+func getParsedYamlTestFixture(t *testing.T) map[string]interface{} {
+	yamlTestString := `
+animals:
```

Comment:
suggestion: This seems too much like we're testing the `[]string` type and not the `map[string][]string` type. I recommend adding at least one more key so it's clear we intend to load all keys from a given level in the configuration hierarchy and load a list for each of them. For example:

```yaml
fish: # A key in the map
  - "dace" # a value in string slice
  - "crappie"
dogs: # another key in the map
  - "pitbull"
  - "chihuahua"
```